### PR TITLE
DocC fixes

### PR DESCRIPTION
**Pull request by stephencelis**
We have a number of outdated DocC references. This PR should hopefully take care of them!

### CODE REVIEW

File: Sources/ComposableArchitecture/TestStore.swift [hunk 1]
```diff
@@ -1043,7 +1043,9 @@ extension TestStore where ScopedState: Equatable {
     let previousState = self.reducer.state
     let task = self.store
       .send(.init(origin: .send(self.fromScopedAction(action)), file: file, line: line))
-    await self.reducer.effectDidSubscribe.stream.first(where: { _ in true })
+    for await _ in self.reducer.effectDidSubscribe.stream {
+      break
+    }
```
Comment (stephencelis) [inline Sources/ComposableArchitecture/TestStore.swift:1048]: Unrelated sendability fix.

File: Sources/ComposableArchitecture/Reducer/Reducers/DebugReducer.swift [hunk 2]
```diff
@@ -6,6 +6,9 @@ extension ReducerProtocol {
   ///
   /// - Parameter printer: A printer for printing debug messages.
   /// - Returns: A reducer that prints debug messages for all received actions.
+  #if swift(>=5.8)
+    @_documentation(visibility: public)
+  #endif
   @inlinable
   public func _printChanges(
```
Comment (stephencelis) [inline Sources/ComposableArchitecture/Reducer/Reducers/DebugReducer.swift:13]: Trying out the new documentation visibility attribute.

**Review by mbrandonw — APPROVED** [review]
